backend/tracim_backend/lib/core/content.py

```diff
@@ -1421,8 +1427,11 @@ def update_content(self, item: Content, new_label: str, new_content: str=None) -
         if not new_label:
             raise EmptyLabelNotAllowed()
 
-        self._is_content_label_free_or_raise(
-            new_label,
+        label = new_label or item.label
+        file_extension = item.file_extension
+        filename = '{}{}'.format(label, file_extension)
```

Comment:
Here you defined a second way to "build" filename. This kind of code result in side effects because the strategy is defined in several places (and maybe different ways). This must be encapsulated and re-used.